File: pool-app-client/src/app/user-pool.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserPoolService {

  private baseUrl = 'http://localhost:8080/springboot-crud-rest/getAllUserPools';

  constructor(private http: HttpClient) { }

  getUserpool(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  createUserpool(userpool: Object): Observable<Object> {
    return this.http.post(`${this.baseUrl}`, userpool);
  }

  updateUserpool(id: number, value: any): Observable<Object> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }


  getUserpoolList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }
}

File: src/main/java/com/rbs/poolApp/services/UserPoolService.java
package com.rbs.poolApp.services;

import com.rbs.poolApp.model.UserPool;

import java.util.List;


public interface UserPoolService {

    UserPool createUserPool (UserPool userPool);

    UserPool updateUserPool (UserPool userPool);

    List< UserPool > getAllUserPools();


}
